### PR TITLE
gh-95588: Drop the safety claim from `ast.literal_eval` docs.

### PR DESCRIPTION
**Pull request by gpshead**
It was never safe and this claim conflicts directly with the big warning in the docs about it being able to crash the interpreter.

Making it safe is not a mere bugfix but an entire rewrite, we're never going to live up to the safety claim via a bugfix and people with existing releases will always have versions around that are unsafe even if we did. So making that claim is bad for our users.

* Issue: gh-95588

### CODE REVIEW

**Comment by njsmith** [comment]
Can we state clearly that maliciously large/complex inputs can cause a DoS (process crash or slow execution time), but nothing worse than that? That's a useful guarantee, and not the same as what most people think of when they think "unsafe for untrusted input".

(Can it even cause unbounded execution time? with the new parser, shouldn't `literal_eval` be O(n) in time and space?)

**Comment by gpshead** [comment]
It only takes 200 bytes to crash the process so it isn't even about being what I'd call large. (which is why i removed the word "large" from the warning)

**Comment by gpshead** [comment]
i'm weary of guaranteeing it doesn't lead to arbitrary code execution. I believe that is the intent and might be true, and any example that allowed that is one we'd care about, but it is hard to _claim_ as the code this uses for parsing wasn't designed with untrusted input in mind because of the rhetorical "who even does that to a compiler?" question.

so the wording i have here tries to explain that it isn't an arbitrary python code execution function. but doesn't specifically say it cannot lead to other things, just that we know you can get memory exhaustion and C stack overflows.

**Comment by gpshead** [comment]
unbounded time is possible but i can't get into the details of that in public right now. at input sizes too small to cause a stack overflow I do not believe you'll run into time issues.

File: Doc/library/ast.rst [hunk 1]
```diff
@@ -1991,20 +1991,27 @@ and classes for traversing abstract syntax trees:
...
+   need to parse the values oneself.  It is not capable of evaluating
+   arbitrarily complex expressions, for example involving operators or
+   indexing.
+
+   This function had been documented as "safe" in the past without defining
+   what that meant. That was misleading. This is specifically designed not to
+   execute Python code, unlike the more general :func:`eval`. There is no
+   namespace, no name lookups, or ability to call out. But it is not free from
+   attack: A relatively small input can lead to memory exhaustion or to C stack
+   exhaustion, crashing the process. Calling it on untrusted input is thus
```
Comment (vstinner) [inline Doc/library/ast.rst]: you may also mention "timeout" (excessive CPU usage?)

File: Doc/library/ast.rst [hunk 2]
```diff
@@ -1991,20 +1991,27 @@ and classes for traversing abstract syntax trees:
...
+   unwise.
 
    .. warning::
-      It is possible to crash the Python interpreter with a
-      sufficiently large/complex string due to stack depth limitations
-      in Python's AST compiler.
+      It is possible to crash the Python interpreter with a complex string due
+      to stack depth limitations in Python's AST compiler.
```
Comment (vstinner) [inline Doc/library/ast.rst]: I don't think that "complex string" is exact here, I suggest:

```suggestion
      It is possible to crash the Python interpreter due
      to stack depth limitations in Python's AST compiler.
```

**Comment by skirpichev** [comment]
Lets try to backport this on 3.9 (see #1267110).

**Pull request merged**